arb-gateway/src/ArbProofService.ts

```diff
@@ -0,0 +1,139 @@
+/* eslint-disable prettier/prettier */
+import { EVMProofHelper, type IProofService } from '@ensdomains/evm-gateway';
+import { AbiCoder, Contract, EventLog, ethers, toBeHex, type AddressLike, toNumber } from 'ethers';
+import rollupAbi from "./abi/rollupABI.js";
+export interface ArbProvableBlock {
+    number: number
+
+    sendRoot: string,
+    blockHash: string,
+    nodeIndex: string,
+    rlpEncodedBlock: string
+}
+
+
+/**
+ * The proofService class can be used to calculate proofs for a given target and slot on the Arbitrum network.
+ * It's also capable of proofing long types such as mappings or string by using all included slots in the proof.
+ *
+ */
+export class ArbProofService implements IProofService<ArbProvableBlock> {
+    private readonly l2Provider: ethers.JsonRpcProvider;
+    private readonly rollup: Contract;
+    private readonly helper: EVMProofHelper;
+
+    constructor(
+        l1Provider: ethers.JsonRpcProvider,
+        l2Provider: ethers.JsonRpcProvider,
+        l2RollupAddress: string,
+
+    ) {
+        this.l2Provider = l2Provider;
+        this.rollup = new Contract(
+            l2RollupAddress,
+            rollupAbi,
+            l1Provider
+        );
+        this.helper = new EVMProofHelper(l2Provider);
+    }
+
+    async getStorageAt(block: ArbProvableBlock, address: AddressLike, slot: bigint): Promise<string> {
+        return this.helper.getStorageAt(block.number, address, slot);
+    }
+
+
+    /**
+     * @dev Fetches a set of proofs for the requested state slots.
+     * @param block A `ProvableBlock` returned by `getProvableBlock`.
+     * @param address The address of the contract to fetch data from.
+     * @param slots An array of slots to fetch data for.
+     * @returns A proof of the given slots, encoded in a manner that this service's
+     *   corresponding decoding library will understand.
+     */
+    async getProofs(
+        block: ArbProvableBlock,
+        address: AddressLike,
+        slots: bigint[]
+    ): Promise<string> {
+        const proof = await this.helper.getProofs(block.number, address, slots);
+
+        return AbiCoder.defaultAbiCoder().encode(
+            [
+                'tuple(bytes32 version, bytes32 sendRoot, bytes32 blockHash,uint64 nodeIndex,bytes rlpEncodedBlock)',
```

Comment:
```suggestion
                'tuple(bytes32 version, bytes32 sendRoot, bytes32 blockHash, uint64 nodeIndex, bytes rlpEncodedBlock)',
```

Reply:
Is it necessary to supply both the block and its hash?

Reply:
good catch, will be removed